rust/src/lib/iface.rs

```diff
@@ -83,82 +104,15 @@ impl Default for InterfaceType {
-            "mac-vlan" => InterfaceType::MacVlan,
-            "mac-vtap" => InterfaceType::MacVtap,
-            "ovs-bridge" => InterfaceType::OvsBridge,
-            "ovs-interface" => InterfaceType::OvsInterface,
-            "veth" => InterfaceType::Veth,
-            "vlan" => InterfaceType::Vlan,
-            "vrf" => InterfaceType::Vrf,
-            "vxlan" => InterfaceType::Vxlan,
-            "infiniband" => InterfaceType::InfiniBand,
-            "tun" => InterfaceType::Tun,
-            "macsec" => InterfaceType::MacSec,
-            "ipsec" => InterfaceType::Ipsec,
-            "unknown" => InterfaceType::Unknown,
-            "xfrm" => InterfaceType::Xfrm,
-            _ => InterfaceType::Other(s.to_string()),
-        }
-    }
-}
-
-impl std::fmt::Display for InterfaceType {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(
-            f,
-            "{}",
-            match self {
-                InterfaceType::Bond => "bond",
-                InterfaceType::LinuxBridge => "linux-bridge",
-                InterfaceType::Dummy => "dummy",
-                InterfaceType::Ethernet => "ethernet",
-                InterfaceType::Loopback => "loopback",
-                InterfaceType::MacVlan => "mac-vlan",
-                InterfaceType::MacVtap => "mac-vtap",
-                InterfaceType::OvsBridge => "ovs-bridge",
-                InterfaceType::OvsInterface => "ovs-interface",
-                InterfaceType::Veth => "veth",
-                InterfaceType::Vlan => "vlan",
-                InterfaceType::Vrf => "vrf",
-                InterfaceType::Vxlan => "vxlan",
-                InterfaceType::InfiniBand => "infiniband",
-                InterfaceType::Unknown => "unknown",
-                InterfaceType::Tun => "tun",
-                InterfaceType::MacSec => "macsec",
-                InterfaceType::Ipsec => "ipsec",
-                InterfaceType::Xfrm => "xfrm",
-                InterfaceType::Other(ref s) => s,
-            }
-        )
-    }
-}
-
-impl Serialize for InterfaceType {
-    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
-    where
-        S: Serializer,
-    {
-        serializer.serialize_str(format!("{self}").as_str())
-    }
-}
-
-impl<'de> Deserialize<'de> for InterfaceType {
-    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
-    where
-        D: Deserializer<'de>,
-    {
-        let v = serde_json::Value::deserialize(deserializer)?;
-        match v.as_str() {
-            Some(s) => Ok(InterfaceType::from(s)),
-            None => Ok(InterfaceType::Unknown),
+impl fmt::Display for InterfaceType {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        if let Ok(interface_type) = serde_yaml::to_string(&self) {
+            let mut interface_type_chars = interface_type.chars();
+            // Remove yaml last new line char
+            interface_type_chars.next_back();
```

Comment:
We are assuming `serde_yaml` not appending `---` at the beginning and always add new line at the end, this is never a API promise of  `serde_yaml`.

This is hack. If ` #[serde(rename` can help your golang binding, then keep both serde rename and `impl std::fmt::Display` with a comment.

Reply:
And I don't like `std::fmt::Display` depend on `serde::Serialize`. The dependency direction is wrong.

Reply:
ok, I will keep both.